ekfNode:
  ros__parameters:
    xPositionDVL: -0.05
    yPositionDVL: 0.2
    zPositionDVL: 0.48
    yawRotationDVL: 2.35619449019

    xPositionIMU: -0.05
    yPositionIMU: 0.11
    zPositionIMU: 0.4

    zPositionDepthSensor: 0.0

rosSlamTest:
  ros__parameters:
    registration_number_of_pixels_dimension: 256 # defines the accuracy and speed of the registration
    registration_size_of_scan: 40.0 # size of the scans. Defined by the sonar(our case 20 m radius)
    registration_threshold_translation: 0.001 # registration threshold. Smaller detects more and smaller peaks(means less noise resistant)
    map_size: 512 #number of pixels in one dimension of the occupancy grid plotted
    dimension_size_map: 45.0 # dimension of the map. In our case its 45 meters. That is enough.
    distance_ignored_around_robot: 1.0 # distance to be ignored around the robot/sonar. Each measurement has a lot of noise in the beginning.
    rotation_sonar_on_robot: 3.14159265359 # rotation of the sonar on top of the robot
    sonar_forward_looking_down: false # false if sonar is put on Top of the robot. True on the bottom(head is then looking down).
    number_of_loop_closures_per_rotation: 1.5 # could be used to make more matches per 360 degree Scan. See Paper for more details.
    measurement_noise_ekf_xy: 0.03 # the x-y measurement noise used in the graph by the estimation of the kalman filter
    measurement_noise_ekf_yaw: 0.03 # the yaw measurement noise used in the graph by the estimation of the kalman filter
    maximum_loopclosure_distance: 5.0 # maximum distance to try a loop closure.(not consecutive scan loop closure)


